### PR TITLE
Updates to files

### LINKED ISSUE #112: Expansion on the excel results file
#### Description:

Expand on the excel results method/class so that block compliance calculations are included within the method.

### PR DESCRIPTION
**Pull request by EKAdebo**
### Issues addressed by this PR

Closes #112 

Excel results methods updated, and parametric components split into files.

### CODE REVIEW

**Comment by EKAdebo** [comment]
@BHoMBot  check compliance

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Please resolve the compliance issues reported

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Comment by EKAdebo** [comment]
@BHoMBot check core
@BHoMBot check serialisation

File: SAP_Engine/Compute/ExcelResults.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        [Description("Takes data such as areas of building parts, and results and then pushes to a single object.")]
+        [Input("filenames", "List of all the files names in a study.")]
+        [Input("reportObjs", "The SAPReport to get the data from.")]
+        [Input("jsonFile", "The JSON file prroduced by running a parametric study.")]
+        [Output("results", "Results pulled from the report.")]
+        public static List<SAPExcelResults> ExcelResults(List<string> filenames, List<SAPReport> reportObjs, BH.oM.Environment.SAP.JSON.JSONReport jsonFile)
+        {
+            //TODO include error messages
+            if (filenames == null || filenames.Count == 0) { return null; }
```
Comment (FraserGreenroyd) [inline SAP_Engine/Compute/ExcelResults.cs]:
```suggestion
            if (filenames == null || filenames.Count == 0 || reportObjs == null || reportObjs.Count == 0 || reportObjs.Count != fileNames.Count)
                return null;
```

File: SAP_Engine/Compute/ExcelResults.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            
+            if (reportObjs == null || reportObjs.Count == 0) { return null; }
+
+            if (reportObjs.Count !=  filenames.Count) { return null; }
+
```
Comment (FraserGreenroyd) [inline SAP_Engine/Compute/ExcelResults.cs]:
```suggestion
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Review by FraserGreenroyd — APPROVED** [review]
All fine now for the time being.

**Pull request merged**